index.js

```diff
@@ -16,62 +23,85 @@ module.exports = class RandomAccessFile extends RandomAccessStorage {
     this.filename = filename
     this.fd = 0
 
-    // makes random-access-storage open in writable mode first
-    if (opts.writable || opts.truncate) this.preferReadonly = false
+    const {
+      readable = true,
+      writable = true
+    } = opts
+
+    this.mode = readable && writable ? RDWR : (readable ? RDONLY : WRONLY)
 
-    this._size = opts.size || opts.length || 0
+    this._size = opts.size || 0
     this._truncate = !!opts.truncate || this._size > 0
     this._rmdir = !!opts.rmdir
-    this._lock = opts.lock || noLock
-    this._sparse = opts.sparse || noLock
+    this._lock = opts.lock === true
+    this._sparse = opts.sparse === true
     this._alloc = opts.alloc || Buffer.allocUnsafe
   }
 
   _open (req) {
     const self = this
+    const mode = this.mode | CREAT
```

Comment:
To make the current tests pass, I had to always open with `O_CREAT` as the tests read from files that haven't yet been created.